sr_wear_logger/src/sr_wear_logger.py

```diff
@@ -0,0 +1,146 @@
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+import rospy
+import rospkg
+import os
+import yaml
+import roslaunch
+import requests
+import shutil
+from sensor_msgs.msg import JointState
+from sr_utilities_common.awsmanager import AWS_Manager
+
+
+class WearLogger():
+
+    def _extractHandData(self, msg):
+        hand_data = dict(zip(msg.name, msg.position))
+        for key in hand_data.keys():
+            if key.startswith("ra") or key.startswith("la"):
+                hand_data.pop(key, None)
+        return hand_data
+
+    def callback(self, msg):
+        hand_data = self._extractHandData(msg)
+        updates = dict.fromkeys(hand_data.keys(), 0)
+
+        if self.first_run:
+            self.previousValues = hand_data
+            self.first_run = False
+
+        for key, value in hand_data.items():
+            valueDifference = abs(self.previousValues[key] - value)
+            if valueDifference > self.threshold:
+                self.currentValues[key] += round(valueDifference, 5)
+        self.previousValues = hand_data
+
+    def _uploadToAWS(self, event):
+        self.aws_manager.upload("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                "data", ["wear_data.yaml"])
+        return rospy.get_param('aws_upload_succeeded')
+
+    def _downloadFromAWS(self):
+        self.aws_manager.download("shadowrobot.benchmarks", rospkg.RosPack().get_path('sr_wear_logger'),
+                                  "data", ["wear_data.yaml"])
+        return rospy.get_param('aws_download_succeeded')
+
+    def _saveDataLocaly(self, event):
+        if self._verifyData():
+            f = open(self.logFilePath+self.logFileName, 'w')
+            completeData = dict()
+            completeData['total_angles'] = self.currentValues
+            completeData['total_time'] = rospy.get_rostime().secs
+            yaml.safe_dump(completeData, f)
+            f.close()
+
+    def __init__(self):
+        self.aws_manager = AWS_Manager()
+        self.first_run = True
+        self.previousValues = {}
+        self.currentValues = {}
+        self.threshold = 0.0175
+        self.logFilePath = "../data/"
+        self.logFileName = "wear_data.yaml"
+        self._initLog()
+        rospy.Timer(rospy.Duration(5), self._saveDataLocaly)
+        rospy.Timer(rospy.Duration(10), self._uploadToAWS)
+        rospy.Subscriber('/joint_states', JointState, self.callback)
+
+    def _verifyData(self):
+        if len(self.currentValues.keys()) == 0:
+            return False
+        return True
+
+    def _loadDataFromYAML(self):
+        f = open(self.logFilePath+self.logFileName, 'r')
+        completeData = yaml.load(f, Loader=yaml.SafeLoader)
+        self.currentValues = completeData['total_angles']
+        self.currentTime = completeData['total_time']
+        f.close()
+
+    def _update_with_bigger(self, dict1, dict2):
```

Comment:
use better name for variables and function. could be: update_with_higher_joint_value. and instead of dict1 and dict2 the actual meaning of those data.

Reply:
Changed